tests/composite_pk/test_filter.py

```diff
@@ -206,6 +206,14 @@ def test_filter_comments_by_pk_in_subquery(self):
             [self.comment_1],
         )
 
+    def test_filter_by_pk_in_subquery_invalid_selected_columns(self):
+        msg = (
+            "The QuerySet value for the 'in' lookup must have 2 selected "
+            "fields (received 3)"
+        )
+        with self.assertRaisesMessage(ValueError, msg):
+            Comment.objects.filter(pk__in=Comment.objects.values("pk", "text"))
```

Comment:
Changing `"text"` to `"user"` unexpectedly still passes. I think this is a separate issue (silently discarding one of the two `ForeignObject` from fields).

Reply:
For me, it does fail (ValueError not raised), unless you're referring to the fact that the query doesn't fail at the db level, which is because I didn't evaluate the queryset. I'm not sure what the house style is on that point when testing errors: some of the tests I added in these modules add `.count()`; others don't.

Reply:
Ah strange, sorry I mussed have messed something up when testing, working for me now